Update java sampling documentation page (#4736)
open-telemetry/opentelemetry.io: content/en/docs/languages/java/sampling.md

needs. In general, decide which sampler to use at the start of a trace and allow
the sampling decision to propagate to other services.

## Default behavior

By default, all spans are sampled, and thus, 100% of traces are sampled. If you

> do not need to manage data volume, don't bother setting a sampler.

Reviewer: ```suggestion
do not need to manage data volume, you don't need to set a sampler.
```

It would be nice to present typical situations representing "manage data volume".

- `traceIdRatioBased`, which samples a fraction of spans, based on the fraction
  given to the sampler. If you set `0.5`, half of all the spans are sampled.
  Currently, only the ratio of traces that are sampled can be relied on, not how

>   the sampled traces are determined. As such, it is recommended to only use this
>   sampler for root spans using `parentBased`.

Reviewer: ```suggestion
  the sampled traces are determined. Only use this sampler for root spans
  that use `parentBased`.
```
